src/util/strprintf.hpp

```diff
@@ -0,0 +1,52 @@
+#pragma once
+
+/* Thin wrapper around std::snprintf for sprintf-like formatting
+ * to std::string. */
+
+#include <cstdio>
+#include <memory>
+#include <string>
+#include <utility>
+#include <vector>
+
+namespace nest {
+namespace mc {
+namespace util {
+
+namespace impl {
+    template <typename X>
+    X sprintf_arg_translate(const X& x) { return x; }
+
+    inline const char* sprintf_arg_translate(const std::string& x) { return x.c_str(); }
+
+    template <typename T, typename Deleter>
+    T* sprintf_arg_translate(const std::unique_ptr<T, Deleter>& x) { return x.get(); }
+
+    template <typename T>
+    T* sprintf_arg_translate(const std::shared_ptr<T>& x) { return x.get(); }
+}
+
+template <typename... Args>
+static std::string strprintf(const char* fmt, Args&&... args) {
+    thread_local static std::vector<char> buffer(1024);
+
+    for (;;) {
+        int n = std::snprintf(buffer.data(), buffer.size(), fmt, impl::sprintf_arg_translate(std::forward<Args>(args))...);
+        if (n<0) {
+            throw std::system_error(errno, std::generic_category());
+        }
+        else if ((unsigned)n<buffer.size()) {
+            return std::string(buffer.data());
+        }
+        buffer.resize(2*n);
+    }
+}
+
+template <typename... Args>
+static std::string strprintf(std::string fmt, Args&&... args) {
```

Comment:
Two comments:
* is static needed?
* this isn't a sink for the `fmt` string, so should it be passed as `const std::string&`?